src/test/test_analyser.py

```diff
@@ -108,10 +108,12 @@ def test_derive_elements(self):
         derived_elements = self.emptyAnalyser.derive_elements(data_obj, outfolder)
         expected = [
             {
+                "id": "el1",
```

Comment:
why exactly are these lines added in this PR? I can't work it out.

Reply:
When the metaanalyser executes the sequence of `analyse_element` calls in its children, it has to work out what `src` and `dest` paths to supply to each child in the element data (these will be locations in the meta analyser's cache). 

It could infer this from the `src` and `dest` values of the previous child, but this would involve splitting the paths into components and making an assumption that they end with the element_id. It seemed cleaner to just include the element_id in the element data passed into analyse_element.